applications/sdp/mspi/src/main.c

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "./hrt/hrt.h"
+
+#include <stdio.h>
+#include <zephyr/kernel.h>
+#include <hal/nrf_vpr_csr.h>
+#include <hal/nrf_vpr_csr_vio.h>
+#include <haly/nrfy_gpio.h>
+
+#include <zephyr/drivers/mspi.h>
+
+#define MAX_DATA_LEN 256
+
+#define HRT_IRQ_PRIORITY	   2
+#define HRT_VEVIF_IDX_WRITE_SINGLE 17
+#define HRT_VEVIF_IDX_WRITE_QUAD   18
+
+#define VEVIF_IRQN(vevif)   VEVIF_IRQN_1(vevif)
+#define VEVIF_IRQN_1(vevif) VPRCLIC_##vevif##_IRQn
+
+struct mspi_config {
+	uint8_t *data;
+	uint8_t data_len;
+	uint8_t word_size;
+};
+
+struct mspi_dev_config {
+	enum mspi_io_mode io_mode;
+	enum mspi_ce_polarity ce_polarity;
+	uint32_t read_cmd;
+	uint32_t write_cmd;
+	uint8_t cmd_length;  /* Command length in bits. */
+	uint8_t addr_length; /* Address length in bits. */
+};
+
+static struct mspi_dev_config mspi_dev_configs;
+
+uint32_t data_buffer[MAX_DATA_LEN + 2];
+
+volatile uint8_t counter_top = 4;
```

Comment:
Magic number?

Reply:
it is magic for now for testing purpouses it will be fixed in next PR. When IPC part is implemented this value will be set according to protocol configuration passed through IPC